### ISSUE TITLE
New People Record Feature (Bond & Insurance)(TL)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1802]
# Work to be performed on #13578

### As a ROW Permit Analyst, I need the RW-TURP folder to validate that the bond & insurance info for the responsible party people record is current.

**Stakeholders:** Paloma, Kim
**DTS Manager:** Tracy

**Acceptance Criteria:**
- **On people type 'Responsible Party', create 9 new people infos**
    - Bond Company Name (text)
    - Bond Status (boolean: Active, Expired, Suspended)
    - Bond Number (text)
    - Bond Expiration Date (date mm/dd/yyyy)
    - Insurance Company Name (test)
    - Insurance Status (boolean: Active, Expired, Suspended)
    - Insurance Policy Number (text)
    - Insurance Effective Date (date mm/dd/yyyy)
    - Insurance Expiration Date (date mm/dd/yyyy)

**On RW, Temporary Right of Way Use:**
- Create new folder issue validation rule(s)
    - People type 'Responsible Party' is on the people tab
    - If 'Responsible Party' people type is found, perform people record info validations:
    1. Insurance & Bond Expiration Dates: If dates are in the past, do not allow folder issuance
    2. Insurance & Bond Status: If either value is not Active, do not allow folder issuance
    - If 'Responsible Party' people type is not found, look for 'Right of Way Contractor'. If found, issue the folder.
    - If 'Right of Way Contractor' people type is not found, look for 'City Department'. If found, issue the folder.
    - If 'City Department' is not found, look for 'Franchised Utility' people type. If found, issue the folder. If not found, do not allow folder issuance.
    - _Stakeholders are open to not performing these validations and instead inserting dummy info for the bond/insurance exceptions like City Departments and Franchised Utility._ 
    3. Shift the Bond/Insurance infos validation to Issue code.
    4.  Bond/Insurance infos must **NOT** be made mandatory

> New AC Items added 8/13/21:
> 
> 1 - PERMIT EXTENSION process - needs validation for BOND/INSURANCE expiry date.
> 2 - VALIDATE Right of Way contractor and Responsible Party on the TURP REVIEW & PERMIT EXTENSION.

 **On Portal; RW-TURP Application, on the People Select Page:**
- Acceptance criteria TBD. The customer needs to provide input for one of the 4 people options (Responsible Party, ROW Contractor, City Department, Franchised Utility).

**Comment by amenity** [comment]
Submitted as a ticket because ROW wasn't sure whether it had to be run past the AMANDA governing board.

**Comment by vschandramouli** [comment]
Had a meeting with Kim and Paloma on 9/18 - confirmed that infos will not be displayed in portal. No additional AC are required for this ticket.

**Comment by Nadin-Nader** [comment]
@TracyLinder This is backlog Ready
@vschandramouli Great job

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader, @TracyLinder - Aishwarya will begin working on this card when # 1923 is completed

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader @TracyLinder - Work is in progress

**Comment by shroffjs** [comment]
@TracyLinder @Nadin-Nader @Sreelatha-BlazeKannan 
A demo(made in localhost) is sent for this ticket in email to you on 10/22/2020.
A Dev deployment request is raised to preserve the code changes in Dev: SCTASK0102012

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder This has been deployed to DEV per Martin

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader @TracyLinder -  Subha will test the same in DEV and update

**Comment by shroffjs** [comment]
@TracyLinder @Nadin-Nader @vschandramouli @Sreelatha-BlazeKannan 
#1802 is deployed in DEV and I tested it. I still see few of new fields added to 'Responsible Party' People type are not enabled to enter/edit the values in DEV also. Me and Nadin are trying to find if any extra privileges are needed to the user login.

Apart from that  I realized the new code is not validating the people status. As of now, with any of these three people type RW-TURP can be issued even the people status is 'inactive'.
So I would like to know that to  issue the permit what are the allowed valid statuses for Responsible Party, City Development and Franchise?

**Comment by TracyLinder** [comment]
We are having permissions issues with this ticket.... We will meet on 11/3/20 to try to figure it out so we can move forward. 
@Nadin-Nader will be doing research on the permissions prior to the meeting.

**Comment by shroffjs** [comment]
**Status:** Tomorrow going to have a call to figure out the permissions issue. 
Apart from that as I mentioned above, 

1. What are the valid status of people type to issue the permit?

**Comment by vschandramouli** [comment]
Had meeting with Kim and Paloma on questions by Subha. It was confirmed that the stakeholders do not want to enforce a people status validation. 

Also updated the following AC: 

  - If 'Responsible Party' people type is not found, look for 'Right of Way Contractor'. If found, issue the folder.
  - If 'Right of Way Contractor' people type is not found, look for 'City Department'. If found, issue the folder.

@SubhaShroff Please refer

**Comment by TracyLinder** [comment]
### Meeting Notes:
Nate will ask Gita if we can add permission to Non-Escrow Permit supervisor & non-escrow permit tech. If she says no, Subha will create new infos to use that will work with current user permissions.

**Comment by TracyLinder** [comment]
@SubhaShroff please move forward with creating new infos that begin with the word ROW. Gita advised Nate that updating permissions would not be an ideal solution.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Currently Subha is working on the changes.

**Comment by shroffjs** [comment]
@TracyLinder @Nadin-Nader @vschandramouli @Sreelatha-BlazeKannan 
Done with the changes and demo video is sent in email today(11/4/2020).

1. Added the 'Right of Way Contractor' people type check to issue permit, in case 'Responsible Party' people type is not found,.
2. Created below mentioned new info fields and added to 'Responsible Party' type(as we discussed on 11/3/2020 to make the info fields enabled.)
ROW Bond Status (boolean: Active, Expired, Suspended)
ROW Bond Expiration Date (date mm/dd/yyyy)
ROW Insurance Status (boolean: Active, Expired, Suspended)
ROW Insurance Effective Date (date mm/dd/yyyy)
ROW Insurance Expiration Date (date mm/dd/yyyy)

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder @Nadin-Nader  - Waiting for feedback

**Comment by TracyLinder** [comment]
### Demo Feedback:
- What will this look like on the people select page on portal?
- Folder issuance on Responsible party type should validate bond status & insurance status
    - If bond or insurance status is not Active, throw an error and do not allow folder to issue

**Comment by TracyLinder** [comment]
### Update
- The AC has been updated to include new validations on insurance & bond status. Also added a section for portal development.
- The portal portion of this work was missed in the original AC. More scoping will be needed & AC to be defined. 

@shroffjs please finish the work you have in progress and promote this to DEV. Thank you!

**Comment by shroffjs** [comment]
@TracyLinder @Nadin-Nader @Sreelatha-BlazeKannan @vschandramouli 
This ticket is deployed in DEV. I tested it for BO requirements and its ready(except the latest Portal portion).

**Comment by Nadin-Nader** [comment]
Take to AOB or find documentation saying its approved by AOB

**Comment by Nadin-Nader** [comment]
This is ready to be Picked after AOB

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder @vschandramouli  Went through the ticket to recollect the facts and review the work done.

1. This ticket Back Office part is in DEV already. I retested it for different people types(RSNs: 11901314, 11901315,  11901317,  11901316). Please let me know if we can move it TEST.
2. AC says portal part TBD. If Portal requirements also added(may be as a separate ticket), I can work on that part too.

**Comment by SubhaShroff** [comment]
The latest demo videos - tested in Amanda BO DEV:
https://user-images.githubusercontent.com/51680715/122281542-fcc06d00-ceaf-11eb-84f1-5b607cdcb3be.mp4

https://user-images.githubusercontent.com/51680715/122281736-372a0a00-ceb0-11eb-933c-e67cb3ae879c.mp4

**Comment by Nadin-Nader** [comment]
@SubhaShroff the demos look good. I will send them to the Stakeholders and see if they approve. If they do, we will get it on TEST, Then we will see if they want to UAT it again before moving to PROD.

**Comment by SubhaShroff** [comment]
### Status: Waiting for feedback from stakeholders with the given demo videos.

**Comment by SubhaShroff** [comment]
### Status:
Waiting on Stakeholder's feedback.

**Comment by Nadin-Nader** [comment]
@SubhaShroff Response from Paloma:

> [Yesterday 2:00 PM] Amayo-Ryan, Paloma
>     Hi Nate!  Yes, I finally did!  A couple of thoughts:
> All of the Info Tab fields should be required.
> 	I didn't see the Folder Info Tab or Folder Process Tab that these records were tied to so I can't say for sure *but *I'm guessing these Folders were set to "Activate Upon Payment" YES.  Which means when the Folder was paid the permit was issued.  Seeing as how this "Activiate Upon Payment" function is up in the air, I think it would be better to tie the system alerts/warning (re: infos missing) to the actual Issuance button ... not sure if that makes sense but maybe we can chat about this tomorrow.

We can discuss more with her at 2pm

**Comment by SubhaShroff** [comment]
### Status:
I need to do modification as per feedback on Friday call with stakeholders.
1. Shift the Bond/Insurance infos validation to Issue code.
2. Bond/Insurance infos must be made mandatory(but its not confirmed yet).

Please update the AC accordingly.

**Comment by Nadin-Nader** [comment]
@Nadin-Nader fix AC to reflect changes

**Comment by Nadin-Nader** [comment]
@SubhaShroff Bond/Insurance infos does not need to be made mandatory.
@Nadin-Nader update AC

**Comment by Nadin-Nader** [comment]
@SubhaShroff updated AC to reflect that Bond/Insurance infos do not need to be made mandatory.

**Comment by SubhaShroff** [comment]
@Nadin-Nader As I was verifying, I realized the Bond/Insurance info validation is actually in Issue code. If I am not wrong, RW TURP can get activated with payment. So when the user tries to pay fee, it calls  issue procedure that will do validations. 

So, please let me know, how you want this to be taken care.

**Comment by TracyLinder** [comment]
**Status:**
In development, pending stakeholder feedback. Currently in DEV.

**Comment by TracyLinder** [comment]
@TracyLinder update w/ new AC: Move the validation from the permit issuance process, and to the approved attempt on 50520 TURP Review process. 
<img width="431" alt="Screen Shot 2021-07-21 at 2 49 46 PM" src="https://user-images.githubusercontent.com/49930868/126551117-bcbb3742-4b2d-48e2-a044-9673ba629d8c.png">

**Comment by SubhaShroff** [comment]
@Nadin-Nader  New changes are done in localhost as per the updated AC. I have sent you the demo in email for your review.

**Comment by SubhaShroff** [comment]
### Status:
Demo of new changes is sent to Nadin in email, for review.

**Comment by Nadin-Nader** [comment]
@SubhaShroff i have reviewed the demo and it looks good to move this to DEV.

## **Approved to go to DEV**

**Comment by SubhaShroff** [comment]
@Nadin-Nader SCTASK0155636 is raised for DEV deployment.

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed to DEV is ready for DEV UAT

**Comment by Nadin-Nader** [comment]
**Status:**
Subha is out until the 4th of August 2021. UAT and Development on this will continue then.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Tested this ticket completely, and it looks good to me on DEV. Kindly, let me know if it can be moved to TEST.

**Comment by Nadin-Nader** [comment]
@SubhaShroff Per Paloma and Kim, this does not need to be on Portal. Lets get this to TEST

**Comment by Nadin-Nader** [comment]
##  **Approved for TEST Deployment🎉🙌‼**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  TEST deployment request is submitted - **SCTASK0158070.**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Tested on TEST post-deployment.

**Comment by SubhaShroff** [comment]
### Status:
Ready for testing on TEST.

**Comment by Nadin-Nader** [comment]
After our meeting and UAT of this on TEST we found that everything was good to go except for the fact that, the TURPS didnt really have a validation for ROW Contractor/Responsible party for Bond and Insurance expirations on the TURP Review and TURP permit Extension processes.
Kim would like for the validation to verify.

New AC Items added:

1 - PERMIT EXTENSION process - needs validation for BOND/INSURANCE expiry date.
2 - VALIDATE Right of Way contractor and Responsible Party on the TURP REVIEW & PERMIT EXTENSION.

**Comment by SubhaShroff** [comment]
### Status:
I will be working to develop the Updated AC, soon after 2397.

**Comment by TracyLinder** [comment]
Subha updated AC. I am moving this back to the backlog until fee updates are over.

**Comment by TracyLinder** [comment]
The updates for this ticket are in TEST and this is causing bugs with issue #6844
**We need to roll this back ASAP**

FYI @Nadin-Nader @SubhaShroff

**Comment by SubhaShroff** [comment]
@Nadin-Nader @TracyLinder Raised a ticket SCTASK0170001 to revert the commits on both DEV & TEST.

**Comment by Nadin-Nader** [comment]
**Status:**
This has been reverted on TEST and DEV.
